presto-common/src/main/java/com/facebook/presto/common/SourceLocation.java

```diff
@@ -11,14 +11,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.facebook.presto.sql.tree;
+package com.facebook.presto.common;
 
-public final class NodeLocation
+public final class SourceLocation
 {
```

Comment:
Maybe we could keep `NodeLocation` as is in presto-parser, which is a very isolated/independent library. And then create a new class `SourceLocation` in common or spi. This is a relatively simple class. We can just copy the column and line numbers from NodeLocation to SourceLocation

Reply:
Why unnecessary duplication of code? Also, I plan to enhance this class to have a stack of source locations that indicate original line numbers before rewrites etc. So it makes sense to have a single class.

Reply:
Oh I see your issue is with introducing dependencies? What if we move it to spi? We really need a good package where we can throw these things. For example, we have a similar issue with Type as well.

Reply:
Ya, for parser, we usually treat it different. presto-parser is a separate library that doesn't depend on any presto internal logic even common or spi. So the dup code might be easier to avoid new dependency